### PR TITLE
Fix regression when bumping Linux to 6.1.88

### PR DESCRIPTION
**Pull request by RinHizakura**
Starting from Linux v6.1.88 commit [f31f521](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.88&id=f31f521ad27e3e476a58c07bf0825c4441317dba), the first bootsec is complete removed and filled with 0xff to reserve for PE header. Since we load the full 512 bytes, those 0xff will break kvm-host. Instead, we only have to take the part of setup header.

### CODE REVIEW

**Comment by RinHizakura** [comment]
Solve #35

File: src/arch/x86/vm.c [hunk 1]
```diff
@@ -131,7 +132,9 @@ int vm_arch_load_image(vm_t *v, void *data, size_t datasz)
     void *kernel = ((uint8_t *) v->mem) + 0x100000;
 
     memset(boot, 0, sizeof(struct boot_params));
-    memmove(boot, data, sizeof(struct boot_params));
+    memmove((void *) boot + offsetof(struct boot_params, hdr),
```
Comment (jserv) [inline src/arch/x86/vm.c]: When using void pointers in calculations, the behavior is undefined.
Consider the following:
```c
    memmove((void *) ((uintptr_t) boot + offsetof(struct boot_params, hdr)),
            (void *) ((uintptr_t) data + offsetof(struct boot_params, hdr)),
            sizeof(struct setup_header));
```

Comment (jserv) [reply src/arch/x86/vm.c]: Add some comments to explain the rationale.

**Comment by jserv** [comment]
Thank @RinHizakura for contributing!

**Pull request merged**